pandas/core/missing.py

```diff
@@ -587,7 +587,7 @@ def interpolate_2d(
     return values
 
 
-def _cast_values_for_fillna(values, dtype: DtypeObj):
+def _cast_values_for_fillna(values, dtype: DtypeObj, has_mask: bool):
```

Comment:
in other places we just *pass* the mask directly, e.g. `mask: Optional[np.ndarray]= None` can we do that here? (i think integer EA ops passing to nanops)

Reply:
no, because just before calling this function we check for mask=None and calculate it in that case (and that check has to come before calling this func)

Reply:
right, so simply pass the mask in, right?

Reply:
No.  The issue is that `mask` and `isna(values)` can be different, in particular for the period-dtype case where i8 values get passed.

I wouldn't worry about this too much.  I mentioned in a recent PR an upcoming implementation of actually-inplace fillna, and that is going to end up getting rid of this function entirely (since it turns out the ensure_float64 branch here is not needed/reached)

Reply:
what's the status on this?

Reply:
my Sep 2020 attempts to implement actually-inplace fillna didnt work out (something in the cython i never managed to track down) but i still think it is viable